### PR TITLE
fix: try fix compute stddev for decimal type overflow

### PR DESCRIPTION
**Pull request by ariesdevil**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

try fix compute stddev for decimal type overflow

- Fixes #[Link the issue here]

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14952)

### CODE REVIEW

**Comment by ariesdevil** [comment]
One of our customer report using decimal(7,2) is going to return a random result, presumably overflow.

Using decimal(36,18) will throw an exception 1049=>Decimal overflow: 10000000000000000000000 * 10000000000000000000000 not in [-]. 

Finally converting the input column type to FLOAT64 outputs the correct result.

After debugging for a while I still don't have any ideas，So I simply covert decimal to float and using an new [stddev algorithm](https://nestedsoftware.com/2018/03/27/calculating-standard-deviation-on-streaming-data-253l.23919.html) that other mainstream databases' used.

@lichuang I don't know much about the original implementation, if you have any idea, feel free to close this PR and modify the original codes. cc @sundy-li

**Comment by sundy-li** [comment]
> After debugging for a while I still don't have any ideas，So I simply covert decimal to float and using an new [stddev algorithm](https://nestedsoftware.com/2018/03/27/calculating-standard-deviation-on-streaming-data-253l.23919.html) that other mainstream databases' used.

It's ok.

I deep look at the implementation of `DecimalNumberAggregateStddevState`, it did not handle the overflow well.

```
 let t = match t.checked_mul(t) {
                Some(t) => t,
                None => {
                    return Err(ErrorCode::Overflow(format!(
                        "Decimal overflow: {} * {} not in [{}, {}]",
                        t,
                        t,
                        T::Scalar::MIN,
                        T::Scalar::MAX,
                    )));
                }
            };
```

**Review by sundy-li — APPROVED** [review]

File: src/query/functions/tests/it/aggregates/testdata/agg_group_by.txt [hunk 1]
```diff
@@ -720,7 +720,7 @@ evaluation (internal):
 | Column | Data                                                                    |
 +--------+-------------------------------------------------------------------------+
 | x_null | NullableColumn { column: UInt64([1, 2, 3, 4]), validity: [0b____0011] } |
-| Output | NullableColumn { column: Float64([NaN, NaN]), validity: [0b______11] }  |
+| Output | NullableColumn { column: Float64([0, 0]), validity: [0b______11] }      |
```
Comment (sundy-li) [inline src/query/functions/tests/it/aggregates/testdata/agg_group_by.txt:723]: seems the result is incorrect.

```duckdb
D select stddev_samp(range) from range(1,3);
┌──────────────────────┐
│ stddev_samp("range") │
│        double        │
├──────────────────────┤
│   0.7071067811865476 │
└──────────────────────┘
```

File: src/query/functions/tests/it/aggregates/testdata/agg_group_by.txt [hunk 2]
```diff
@@ -710,7 +710,7 @@ evaluation (internal):
 | Column | Data                                                                                    |
 +--------+-----------------------------------------------------------------------------------------+
 | dec    | NullableColumn { column: Decimal128([1.10, 2.20, 0.00, 3.30]), validity: [0b____1011] } |
-| Output | NullableColumn { column: Decimal128([0.0000, 0.7778]), validity: [0b______11] }         |
+| Output | NullableColumn { column: Float64([0, 0.7778174593]), validity: [0b______11] }           |
```
Comment (sundy-li) [inline src/query/functions/tests/it/aggregates/testdata/agg_group_by.txt:713]:
```
D select stddev_samp(a) from   (values ( 1.1),(2.2),(3.3 ) ) t(a);
┌────────────────────┐
│   stddev_samp(a)   │
│       double       │
├────────────────────┤
│ 1.0999999999999999 │
```

**Comment by sundy-li** [comment]
The previous pr of stddev.

 generate wrong answer in number, should be 1.0
```
databend-local:) select stddev(a) from   (values (1),(2), (3) ) t(a);
┌──────────────┐
│   stddev(a)  │
│ Float64 NULL │
├──────────────┤
│ 0.8164965809 │
└──────────────┘
1 row result in 0.002 sec. Processed 0 rows, 0 B (0 rows/s, 0 B/s)
```

wrong answer in decimal, should be `1.0999999999999999`
```
databend-local:) select stddev(a) from   (values ( 1.1),(2.2),(3.3 ) ) t(a);
┌─────────────────────┐
│      stddev(a)      │
│ Decimal(38, 4) NULL │
├─────────────────────┤
│ 0.0089              │
└─────────────────────┘
1 row result in 0.005 sec. Processed 0 rows, 0 B (0 rows/s, 0 B/s)
```


wrong answer in float64,  should be `1.0999999999999999`
```
databend-local:) select stddev(a) from   (values ( 1.1::Float64),(2.2::Float64),(3.3::Float64 ) ) t(a);
┌──────────────┐
│   stddev(a)  │
│ Float64 NULL │
├──────────────┤
│ 0.898146239  │
└──────────────┘
1 row result in 0.009 sec. Processed 0 rows, 0 B (0 rows/s, 0 B/s)
```

**Comment by ariesdevil** [comment]
This is not a bug, we have previously been mapping `stddev` to `stddev_pop`, MySQL does same mapping, but pg and duckdb mapping `stddev` to `stddev_samp`.

**Review by lichuang — APPROVED** [review]

**Pull request merged**